xapian-core/cluster/cluster.cc

```diff
@@ -0,0 +1,391 @@
+	add_document(it.get_document());
+    docs_num = docs.size();
+}
+
+doccount
+DummyFreqSource::get_termfreq(const string &) {
+    LOGCALL(API, doccount, "DummyFreqSource::get_termfreq()", NO_ARGS);
+    return 1;
+}
+
+doccount
+DummyFreqSource::get_doccount() const {
+    LOGCALL(API, doccount, "DummyFreqSource::get_doccount()", NO_ARGS);
+    return 1;
+}
+
+void
+TermListGroup::add_document(const Document &document) {
+    LOGCALL_VOID(API, "TermListGroup::add_document()", document);
+
+    TermIterator titer(document.termlist_begin());
+    TermIterator end(document.termlist_end());
+
+    for (; titer != end; ++titer) {
+	unordered_map<string, doccount>::iterator i;
+	i = termfreq.find(*titer);
+	if (i == termfreq.end())
+	    termfreq[*titer] = 1;
+	else
+	    i->second += 1;
+    }
+}
+
+doccount
+TermListGroup::get_doccount() const {
+    LOGCALL(API, doccount, "TermListGroup::get_doccount()", NO_ARGS);
+    return docs_num;
+}
+
+doccount
+TermListGroup::get_termfreq(const string &tname) {
+    LOGCALL(API, doccount, "TermListGroup::get_termfreq()", tname);
+    return termfreq[tname];
+}
+
+int
+DocumentSet::size() const {
+    LOGCALL(API, int, "DocumentSet::size()", NO_ARGS);
+    return docs.size();
+}
+
+void
+DocumentSet::add_document(Document doc) {
+    LOGCALL_VOID(API, "DocumentSet::add_document()", doc);
+    docs.push_back(doc);
+}
+
+Document
+DocumentSet::operator[](doccount i) {
+    return docs[i];
+}
+
+class PointTermIterator : public TermIterator::Internal {
+    unordered_map<string, double>::const_iterator i;
+    unordered_map<string, double>::const_iterator end;
+    termcount size;
+    bool started;
+  public:
+    PointTermIterator(const unordered_map<string, double> &termlist) :
+    i(termlist.begin()), end(termlist.end()), size(termlist.size()), started(false)
+    {}
+    termcount get_approx_size() const { return size; }
+    termcount get_wdf() const { throw UnimplementedError("PointIterator doesn't support get_wdf()"); }
+    string get_termname() const { return i->first; }
+    doccount get_termfreq() const { throw UnimplementedError("PointIterator doesn't support get_termfreq()"); }
+    Internal * next();
+    termcount positionlist_count() const {
+	 throw UnimplementedError("PointTermIterator doesn't support positionlist_count()");
+    }
+    bool at_end() const;
+    PositionIterator positionlist_begin() const {
+	throw UnimplementedError("PointTermIterator doesn't support positionlist_begin()");
+    }
+    Internal * skip_to(const string &term);
+};
+
+TermIterator::Internal *
+PointTermIterator::next() {
+    if (!started) {
+	started = true;
+	return NULL;
+    }
+    Assert(i != end);
+    ++i; return NULL;
+}
+
+bool
+PointTermIterator::at_end() const
+{
+    if (!started) return false;
+    return i == end;
+}
+
+TermIterator::Internal *
+PointTermIterator::skip_to(const string &term) {
+    if (i->first == term)
+	return NULL;
+    while (i->first != term)
+	i++;
```

Comment:
Oh yes, definitely. I didn't think about that when I wrote it. But in any case, we do not use the skip_to method in the code. So do I keep it unimplemented for now or do I change this to do a lookup?